src/services/storage/types/url.ts

```diff
@@ -6,9 +6,30 @@ export interface StoredUrl {
 }
 
 export interface UrlStorageDriver {
-	get(id: string): Promise<StoredUrl>
-	save(url: string): Promise<StoredUrl>
+	get(id: string, options: { withInfo: boolean }): Promise<StoredUrl | UrlWithInformation>
+
+	save(url: UrlRequestData): Promise<StoredUrl>
+
 	edit(id: string, url: string): Promise<StoredUrl>
+
 	delete(id: string): Promise<void>
+
 	deleteOverdue(timespanMs: number): Promise<number>
+
+	incVisitCount(id: string): Promise<void>
+
+	incInfoCount(id: string): Promise<void>
+}
+
+export interface UrlRequestData {
+	url: string
+	ip: string
+}
+
+export interface UrlWithInformation extends StoredUrl {
+	urlId: string
+	ip: string
+	urlVisitCount: number
+	infoVisitCount: number
+	lastUsed: string
```

Comment:
I see you've changed it here, but there are other places not changed to lastUsed

Reply:
fixed

Reply:
Not everywhere. Can you recheck?

Reply:
pushed